DOCS userhelp section for trusted devices

brynwhyman opened the pull request:
Related to: https://github.com/silverstripe/silverstripe-mfa/issues/384

Commit:
DOCS userhelp section for trusted devices

robbieaverill reviewed (changes requested):
There's a space in your filename, can you remove that?

ScopeyNZ commented:
I'm guessing this has come up probably because dev agencies have built sites that opt out of having that "remember me" feature. Do you want to put some statement in that indicates this feature might not be available on all sites?

robbieaverill commented:
I think it's worth mentioning here that this is not the same as "don't ask again on this computer" which would still require you to login but wouldn't prompt you for MFA if it remembers your machine (via cookies)

brynwhyman commented:
Thanks for the feedback, I'll add those points. 

> this is not the same as "don't ask again on this computer"

Yes, I'm really raising this PR as a result of [that not being available](https://github.com/silverstripe/silverstripe-mfa/issues/384), so the distinction should probably be noted.

brynwhyman commented:
I think ideally, a "don't ask again on this computer" - type feature is built into the s[ecurity-extensions](https://github.com/silverstripe/silverstripe-security-extensions) module to override the core 'remember me' functionality if MFA is present

brynwhyman commented:
Updated file name and contents based on the comments.

robbieaverill commented:
I agree but think we should put that feature into the MFA module since it's part of MFA

sachajudd commented on docs/en/userguide/01_User_manual/06_Remember_trusted_devices.md:
```diff
@@ -0,0 +1,30 @@
+
+# Remember trusted devices
+
+In order to avoid repeat requests to log in with multi-factor authentication through a device you know to be secure, it is possible to assign your device as a 'trusted device' at the log-in screen.
```
```suggestion
In order to avoid repeat requests to log in with Multi-factor authentication (MFA) through a device you know to be secure, it is possible to assign your device as a 'trusted device' at the log-in screen.
```

sachajudd commented on docs/en/userguide/01_User_manual/06_Remember_trusted_devices.md:
```diff
@@ -0,0 +1,30 @@
+
+# Remember trusted devices
+
+In order to avoid repeat requests to log in with multi-factor authentication through a device you know to be secure, it is possible to assign your device as a 'trusted device' at the log-in screen.
```
```suggestion
In order to avoid repeat requests to log in with MFA through a device you know to be secure, it is possible to assign your device as a 'trusted device' at the log in screen.
```

sachajudd replied:
Since the main document explains what Multi-factor authentication already is we can just use the acronym.

sachajudd replied:
In this case I think possibly 'log in' would be the correct terminology?

sachajudd commented on docs/en/userguide/01_User_manual/06_Remember_trusted_devices.md:
```diff
@@ -0,0 +1,30 @@
+
+2. In addition to entering your Email and Password, choose the option to **Keep me signed in**.
+
+3. Continue to log in with your registered multi-factor authentication method.
```
```suggestion
3. Continue to log in with your registered MFA method.
```

sachajudd commented on docs/en/userguide/01_User_manual/06_Remember_trusted_devices.md:
```diff
@@ -0,0 +1,30 @@
+A device shared between multiple people would not be considered secure and you should not use this option.
+</div>
+
+This functionality is provided by default with the SilverStripe CMS however it may be disabled by a Developer in some projects.
```
```suggestion
This functionality is provided by default with Silverstripe CMS however it may be disabled by a Developer in some projects.
```

sachajudd commented on docs/en/userguide/01_User_manual/06_Remember_trusted_devices.md:
```diff
@@ -0,0 +1,30 @@
+
+## Marking you current device as trusted
+
+1. Visit the log-in screen.
```
```suggestion
1. Visit the log in screen.
```

sachajudd commented on docs/en/userguide/01_User_manual/06_Remember_trusted_devices.md:
```diff
@@ -0,0 +1,30 @@
+<div class="notice" markdown='1'>
+It is important that you only choose to use this option on a device you know to be secure. A secure device may be one that requires you to log in before being able to access the Internet, for example one within an office network, or a personal mobile device with a passcode.
+
+A device shared between multiple people would not be considered secure and you should not use this option.
```
```suggestion
**A device shared between multiple people would not be considered secure and you should not use this option.**
```

sachajudd reviewed (changes requested):
Looks good, just some minor suggestions 🙂

Commit:
DOCS feedback on MFA userhelp

brynwhyman commented:
I've made another commit that's addressed some of your feedback @sachajudd.
* Updated "multi-factor authentication" to "MFA"

I've also made the following changes:
* Updated "log-in screen" to "login screen" based on our [internal writing style guide](https://silverstripe.atlassian.net/wiki/spaces/PROD/pages/506364107/Silverstripe+Writing+Style+Guide)

I haven't made the change from "SilverStripe" to "Silverstripe" as it seems like the docs should reflect the status quo of "SilverStripe" and instead leave updates to do this in one go.

sachajudd commented:
> I haven't made the change from "SilverStripe" to "Silverstripe" as it seems like the docs should reflect the status quo of "SilverStripe" and instead leave updates to do this in one go.

User help has all been updated to use 'Silverstripe CMS'. We could do it all in one go for this particular module but I think it's better to change. I'll create a separate PR updating all of the SilverStripes this week in this module but one won't make a big difference for now.
